src/ucm/mmap/install.c

```diff
@@ -142,6 +132,24 @@ static ucs_status_t ucm_mmap_test(int events)
     return UCS_OK;
 }
 
+/* Called with lock held */
+static ucs_status_t ucm_mmap_test(int events)
+{
+    static int installed_events = 0;
+    ucs_status_t status;
+
+    if (ucs_test_all_flags(installed_events, events)) {
+        /* All requested events are already installed */
+        return UCS_OK;
+    }
+
+    status = ucm_test_events(events);
+    if (status == UCS_OK) {
+        installed_events |= events;
```

Comment:
We do not install events partially now. Is it ok?

Reply:
yes, this call doesn't install hooks at all - just trying catch events on system API call

Reply:
yes, but you will not cache added events, like it was done before. So the next call with less events will go thru the whole check as well

Reply:
right, but this way simplifier logic :)

Reply:
restored original algo